lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1402,27 +1403,34 @@ protected function getSelectManyToManyJoinSQL(array $manyToMany)
         return ' INNER JOIN ' . $joinTableName . ' ON ' . implode(' AND ', $conditions);
     }
 
+    public function getMaxBatchedInsertQueries()
+    {
+        $idGenerator    = $this->class->idGenerator;
+        $isPostInsertId = $idGenerator->isPostInsertGenerator();
+
+        if ($isPostInsertId || $this->class->isVersioned) {
+            return 1;
+        }
+
+        return 50;
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function getInsertSQL()
+    public function getInsertSQL(int $recordsCount = 1)
     {
-        if ($this->insertSql !== null) {
-            return $this->insertSql;
-        }
```

Comment:
So now if we call this method several times in a row and there is no batching, we will re-compute the insertSQL every time?

Reply:
This is correct, I was not able to measure any significant performance difference. Also, other queries like UPDATE and DELETE also rebuild the query from scratch every time.

Reply:
It would be helpful if you can provide the method how you run the performance tests to see any improvements. If this PR gets merged, we will need a way to prevent regressions in the future that will drop any performance improvements of this.

Reply:
I completely agree that full insert SQL cache is not possible, however, we should be able to store columns and placeholders to prevent us from recomputing every time. I recall a 2s slowdown in the whole testing suite by dropping out this cache.
Actually, now that I think about it... we should still store the full insert SQL referring to a single entry. Also, we should store the column count, so we can easily build new sets of placeholders.

Now for update and delete, the reason why they are not cached is because we're not doing full updates. If we were to be more inline with EclipseLink, Hibernate and EntityFramework, we would do full updates (where you update all columns, excluding LOB fields) and also cache this computation. For delete, id based removals would be ok too...
But this is another micro-optimization that could be done in a separate PR.